Add cluster upgrade support by including MOFED dependency in RDMA scenarios

tginer opened the pull request:
This commit adds the proper GPU driver wait for the MOFED driver to be ready so RDMA APIs are available when driver is recompiled. This ensures the operator supports cluster upgrades, i.e., kernel version upgrades.

tginer committed:
Add cluster upgrade support by including MOFED wait nodeSelector in RDMA scenarios.

Signed-off-by: Teresa Giner <tginer@redhat.com>

tariq1890 commented on controllers/object_controls.go:
```diff
@@ -1044,6 +1044,14 @@ func TransformDriver(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec, n C
+		if obj.Spec.Template.Spec.NodeSelector == nil {
+			obj.Spec.Template.Spec.NodeSelector = make(map[string]string)
+		}
+		obj.Spec.Template.Spec.NodeSelector["network.nvidia.com/operator.mofed.wait"] = "false"
```
This was the [commit](https://github.com/Mellanox/network-operator/commit/e49c4d1a35880fc4d76fd57a4ed5ce9b8a06cb2d) that introduced this label. So it's been around for a long time

tariq1890 commented on controllers/object_controls.go:
```diff
@@ -1044,6 +1044,14 @@ func TransformDriver(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec, n C
 	}
+
+	// add nodeSelector for MOFED wait label when GPUDirect RDMA is enabled
+	if config.Driver.GPUDirectRDMA != nil && config.Driver.GPUDirectRDMA.IsEnabled() {
```
```suggestion
	if config.Driver.GPUDirectRDMA != nil &&
		config.Driver.GPUDirectRDMA.IsEnabled() &&
		!config.Driver.GPUDirectRDMA.IsHostMOFED() {
```

tginer committed:
Exclude useHostMofed scenario

Ensure Network Operator label is read only when it is installed; excluding those cases where MOFED is independently installed

Co-authored-by: Tariq <tariq181290@gmail.com>

rajatchopra commented on controllers/object_controls.go:
```diff
@@ -1044,6 +1044,14 @@ func TransformDriver(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec, n C
+		if obj.Spec.Template.Spec.NodeSelector == nil {
+			obj.Spec.Template.Spec.NodeSelector = make(map[string]string)
+		}
+		obj.Spec.Template.Spec.NodeSelector["network.nvidia.com/operator.mofed.wait"] = "false"
```
I wish the label would come from the network operator library. And not magic coded.
